flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/DeclarativeSlotManager.java

```diff
@@ -647,7 +647,10 @@ private ResourceCounter tryFulfillRequirementsWithPendingSlots(
                     pendingSlots = allocationResult.getNewAvailableResources();
                     if (!allocationResult.isSuccessfulAllocating()
                             && sendNotEnoughResourceNotifications) {
-                        LOG.warn("Could not fulfill resource requirements of job {}.", jobId);
+                        LOG.warn(
```

Comment:
Unrelated to this PR: I am not sure whether not enough resource availability should be a warning because this is a normal condition in Flink and does not indicate a problem from Flink's perspective.